app/src/main/java/fr/free/nrw/commons/nearby/NearbyFilterSearchRecyclerViewAdapter.java

```diff
@@ -120,6 +121,8 @@ protected FilterResults performFiltering(CharSequence constraint) {
                 }
 
                 if (constraint == null || constraint.length() == 0) {
+                    // set my bookmarks as the first item
+                    Collections.swap(labels, 0 , labels.indexOf(Label.fromText("BOOKMARK")));
```

Comment:
So this should be completely unnecessary, enums are inherently ordered.

This is the culprit in Label, mucking up ordering
```
public static final Map<String, Label> TEXT_TO_DESCRIPTION
            = new HashMap<>(Label.values().length);

    static {
        for (Label label : values()) {
            TEXT_TO_DESCRIPTION.put(label.text, label);
        }
    }
```

This map doesn't need to exist, the only usage is when constructing this class. A list can be created out of `Label.values()`

Also you could have directly referenced the bookmark eg `labels.indexOf(Label.BOOKMARK)`

Reply:
Hmm, but this map is actually useful to get label from text:
```
        Label label = TEXT_TO_DESCRIPTION.get(text);
```

Reply:
tch, I must have had read access off in my find usages without realising.

TEXT_TO_DESCRIPTION isn't needed as the input to this adapter because `values` suits just fine and we don't need to do a swap